test/e2e/set/scyllacluster/scyllacluster_pv.go

```diff
@@ -5,35 +5,254 @@ package scyllacluster
 import (
 	"context"
 	"fmt"
+	"strings"
+	"sync"
 
 	g "github.com/onsi/ginkgo/v2"
 	o "github.com/onsi/gomega"
 	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
 	"github.com/scylladb/scylla-operator/pkg/controllerhelpers"
+	"github.com/scylladb/scylla-operator/pkg/helpers"
 	"github.com/scylladb/scylla-operator/pkg/naming"
+	"github.com/scylladb/scylla-operator/pkg/pointer"
 	scyllafixture "github.com/scylladb/scylla-operator/test/e2e/fixture/scylla"
 	"github.com/scylladb/scylla-operator/test/e2e/framework"
 	"github.com/scylladb/scylla-operator/test/e2e/utils"
+	"github.com/scylladb/scylla-operator/test/e2e/utils/image"
 	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	watchtools "k8s.io/client-go/tools/watch"
+	"k8s.io/component-helpers/storage/volume"
 )
 
-var _ = g.Describe("ScyllaCluster Orphaned PV", func() {
-	defer g.GinkgoRecover()
-
+var _ = g.Describe("ScyllaCluster Orphaned PV controller", func() {
 	f := framework.NewFramework("scyllacluster")
 
+	const cloneLabelKey = "e2e.operator.scylladb.com/orphaned-pv-test"
+
+	g.JustBeforeEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		testStorageClass := &storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   f.Namespace(),
+				Labels: f.CommonLabels(),
+			},
+			Provisioner:       volume.NotSupportedProvisioner,
+			ReclaimPolicy:     pointer.Ptr(corev1.PersistentVolumeReclaimDelete),
+			VolumeBindingMode: pointer.Ptr(storagev1.VolumeBindingWaitForFirstConsumer),
+		}
+
+		_, err := f.KubeAdminClient().StorageV1().StorageClasses().Create(ctx, testStorageClass, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.JustAfterEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		scErr := f.KubeAdminClient().StorageV1().StorageClasses().Delete(ctx, f.Namespace(), metav1.DeleteOptions{})
+		pvErr := f.KubeAdminClient().CoreV1().PersistentVolumes().DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+			LabelSelector: labels.SelectorFromSet(map[string]string{
+				cloneLabelKey: f.Namespace(),
+			}).String(),
+		})
+
+		o.Expect(scErr).NotTo(o.HaveOccurred())
```

Comment:
Putting the error check far away is a bit error prone. Any reason not to use `utilerrors.NewAggregate` and handle it right after the function call?

Reply:
no reason, done